### ISSUE TITLE
`ResourceId` opaque type cannot be used in places that require `String` easiliy

### ISSUE THREAD

**Issue by pawelprazak** [VirtusLab/besom#145]
```
    [error] Found:    besom.internal.Output[besom.types.ResourceId]
    [error] Required: besom.internal.Input[String | String]
    [error]       restApi = api.id,
    [error]                 ^^^^^^
```

Instead of this:
```
restApi = api.id.map(_.asString), // FIXME this is a hack
```

It would be nice to be able to use this:
```
restApi = api.id,
```

or, where applicable, just this:
```
restApi = api,
```

Related to #144

**Comment by pawelprazak** [comment]
As a side note, I'm not sure why `bucket = feedBucket.id.asString,` didn't work as a workaround.

**Comment by lbialy** [comment]
@prolativ can we generate this as `opaque type ResourceId <: String = String`?

**Comment by lbialy** [comment]
also good question regarding `.asString` extension method

**Comment by lbialy** [comment]
Ah, it's *that* ResourceId! Hmm, ok, we could use `<: NES` (it's also a `<:
String`). I just hope we will never get an empty string in that field (this
should never happen). Why was it in an option though?

**Comment by lbialy** [comment]
Ah, it was in Output. Ok, it will work with Input[String] then I think. No
need for extension method.

**Comment by lbialy** [comment]
Solved by #170
